src/zalsa_local.rs

```diff
@@ -309,7 +309,7 @@ pub(crate) struct QueryRevisions {
     ///   previous revision. To handle this, `diff_outputs` compares
     ///   the structs from the old/new revision and retains
     ///   only entries that appeared in the new revision.
-    pub(super) tracked_struct_ids: IdentityMap,
+    pub(super) tracked_struct_ids: Option<Box<IdentityMap>>,
```

Comment:
```suggestion
    ///
    /// Since not all queries produce a tracked struct, wrapping 
    /// `IdentityMap` in an `Option<Box<T>>` reduces the size of
    /// `QueryRevisions` 3 words (24 bytes on a 64-bit platform).
    pub(super) tracked_struct_ids: Option<Box<IdentityMap>>,
```